src/components/FeatureViews/AdminPortal/BulkUploadModal.js

```diff
@@ -0,0 +1,523 @@
+                                    >
+                                        Courses
+                                    </MenuItem>
+                                    <MenuItem
+                                        value='Enrollments'
+                                        ListItemClasses={{
+                                            selected: classes.menuSelected,
+                                        }}
+                                        className={classes.menuSelect}
+                                    >
+                                        Course Enrollments
+                                    </MenuItem>
+                                </Select>
+                                <IconButton
+                                    disabled={!template && true}
+                                    onClick={handleDownloadTemplate}
+                                >
+                                    <SvgIcon>
+                                        <path
+                                            d='M17.5 13.75V17.5H2.5V13.75H0V17.5C0 18.875 1.125 20 2.5 20H17.5C18.875 20 20 18.875 20 17.5V13.75H17.5ZM16.25 8.75L14.4875 6.9875L11.25 10.2125V0H8.75V10.2125L5.5125 6.9875L3.75 8.75L10 15L16.25 8.75Z'
+                                            fill={template ? omouBlue : gloom}
+                                        />
+                                    </SvgIcon>
+                                </IconButton>
+                            </div>
+
+                            <Grid style={{ textAlign: 'right' }} item xs={12}>
+                                <ResponsiveButton
+                                    style={{ border: 'none', color: darkGrey }}
+                                    variant='outlined'
+                                    onClick={closeModal}
+                                >
+                                    cancel
+                                </ResponsiveButton>
+                                <ResponsiveButton
+                                    style={{
+                                        border: 'none',
+                                        background: white,
+                                    }}
+                                    disabled={!template && true}
+                                    variant={
+                                        template ? 'outlined' : 'contained'
+                                    }
+                                    template={template}
+                                    onClick={handleStepChange}
+                                >
+                                    continue to upload
+                                </ResponsiveButton>
+                            </Grid>
+                        </Grid>
+                    </Grid>
+                );
+            case 1:
+                return (
+                    <>
+                        {loading ? (
```

Comment:
Why is the loading logic handled here instead of at the component level before the return statement?

Reply:
adjust the loading component to handle this case
```
<Loading isLoading={loading}>
    <UploadTemplateStep/>
</Loading>
```